src/Fhir.Metrics/Utils/SearchExtensions.cs

```diff
@@ -18,29 +18,36 @@ public static class SearchExtensions
     {
         ///<summary>
         /// Creates a string from a decimal that allows compare-from-left string searching 
-        /// for finding values that fall within a the precision of a given string representing a decimal .
+        /// for finding values that fall within a the precision of a given string representing a decimal.
+        /// Each Digit caries the significance of the original digits more to the right.
         ///</summary>
-        private static string LeftSearchableNumberString(string s)
+        private static string LeftSearchableNumberString(string original, int exponent)
         {
-            StringBuilder b = new StringBuilder(s);
+            StringBuilder leftSearchable = new StringBuilder(original);
             int reminder = 0;
 
-            for (int i = b.Length - 1; i >= 0; i--)
+            for (int i = leftSearchable.Length - 1; i >= 0; i--)
             {
-                if (b[i] == '.') continue;
-                int n = (int)Char.GetNumericValue(b[i]);
+                if (leftSearchable[i] == '.') continue; // End of the decimal fraction part of the quantity
+
+                // Add the reminder to the number on the left side
+                int n = (int)char.GetNumericValue(leftSearchable[i]);
                 n += reminder;
 
-                reminder = n / 10;
+                reminder = n / 10; // Reset the reminder to 0 in case n < 10
                 n = n % 10;
-                reminder += (n > 5) ? 1 : 0;
-                char c = Convert.ToString(n)[0];
-                b[i] = c;
+                if (n > 5)
+                    reminder += 1;
 
+                leftSearchable[i] = Convert.ToString(n)[0];
             }
-            return b.ToString();
-        }
 
+            // Check if we still have a reminder that we need to carry to the left
+            if (reminder != 0 && leftSearchable[0] == '0')
+                exponent++;
+
+            return $"E{exponent}x{leftSearchable}";
```

Comment:
What if exponent >9 (that is: longer than 1 digit)?
Suggest to simply output 2 digits and error on exponent > 99.